### DISCUSSION TITLE
How do I use GitHub API for getting a list of all changes in a PR?

### DISCUSSION THREAD

**Discussion by Aaron-Garrett** [community/community#79111] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

I am wanting to automate some code checks, but only on lines of code that have been changed due to different approvers over time and not wanting false errors for old code. What this looks like in practice can vary: creating a new file from the lines of code, creating a set or list, etc. To be more specific, I am only wanting lines that have been added. For example:

Resources:
   Type: AWS::IAM::Role
Then that gets updated to:
Resources:
    Type: AWS::IAM::Role
    Name: testing-role

I would want _only_ the line Name: testing-role. I am using Python for the checks, so using PyGithub would be great, but if it is something that can be done in a GitHub Actions yml file, that is great too!

TL;DR: Need a file/list of all lines of code that were changed in a PR by using either PyGithub or GitHub API.

Any help would be appreciated!

**Comment by fury-05** [comment]
You can use the GitHub API to fetch the pull request details and then identify the changed lines in each file. Python script using the PyGithub library that you can use for this purpose, this will fetch the patch content for each file in the pull request. It then identifies the added lines and stores them customize script further based on your specific requirements to achieve desired output.

Create a new test repo for this after achieving desired result then apply the logic to main project.

**Reply by Aaron-Garrett** [reply]
How do I identify the changed lines? I know about the GitHub API to fetch the details. Is there any chance you could share some sample code?

**Reply by fury-05** [reply, marked as answer]
Well from main question you are using AWS so it will require help from expert as to use api and share results and get desired output for your project.

Here is a general sample code you can refer 


```python
from github import Github

//Replace 'your_access_token' with your GitHub personal access token
g = Github("your_access_token")

//Replace 'repo_owner' and 'repo_name' with the owner and name of the repository
repo_owner = 'repo_owner'
repo_name = 'repo_name'
repo = g.get_repo(f"{repo_owner}/{repo_name}")

//Iterate through all open pull requests in the repository
for pull_request in repo.get_pulls(state='open'):
    print(f"Checking Pull Request #{pull_request.number} - {pull_request.title}")

    //Get the list of files changed in the pull request
    files = pull_request.get_files()

   //Iterate through each file and print the added lines
    for file in files:
        print(f"File: {file.filename}")

        //Fetch the content of the patch for the file
        patch_content = file.patch

        //Split the patch content into lines
        patch_lines = patch_content.split('\n')

        //Identify and print added lines
        added_lines = [line[2:] for line in patch_lines if line.startswith('+ ') and not line.startswith('+++')]
        print("Added Lines:")
        for added_line in added_lines:
            print(added_line)

        print("\n" + "=" * 50 + "\n")  # Separating each file's output

    print("\n" + "=" * 50 + "\n")  # Separating each pull request's output
```


## This is not your solution just a example how can this be used to achieve your final goal.

**Reply by Aaron-Garrett** [reply]
Awesome! Thanks so much! I think this is very close to what I am looking for. Everything is running within GitHub Actions before any connection to AWS. Thank you so much.
